src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+
+  const timeout = setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 && !resolved) {
+      clearTimeout(timeout);
+      resolved = true;
+      resolve('First promise was resolved');
+    }
+  });
```

Comment:
You are adding an event listener inside the promise executor. This could lead to memory leaks if the promise is never resolved or rejected. Consider moving the event listener outside the promise and use a variable to store the state of the click.